src/ompl/geometric/planners/rrt/src/STRRTstar.cpp

```diff
@@ -0,0 +1,1114 @@
+
+    OMPL_INFORM("%s: Starting planning with time bound factor %.2f", getName().c_str(), newBatchTimeBoundFactor);
+
+    while (!ptc)
+    {
+        numIterations_++;
+        TreeData &tree = startTree ? tStart_ : tGoal_;
+        tgi.start = startTree;
+        startTree = !startTree;
+        TreeData &otherTree = startTree ? tStart_ : tGoal_;
+
+        // batch is full
+        if (!isTimeBounded_ && (unsigned int)numBatchSamples >= batchSize)
+        {
+            if (firstBatch)
+            {
+                firstBatch = false;
+                oldBatchSampleProb = 0.5 * (1 / timeBoundFactorIncrease_);
+            }
+            oldBatchTimeBoundFactor = newBatchTimeBoundFactor;
+            newBatchTimeBoundFactor *= timeBoundFactorIncrease_;
+            startTree = true;
+            if (sampleUniformForUnboundedTime_)
+                batchSize = std::ceil(2.0 * (timeBoundFactorIncrease_ - 1.0) *
+                                      static_cast<double>(tStart_->size() + tGoal_->size()));
+            numBatchSamples = 0;
+            if (!newBatchGoalMotions_.empty())
+            {
+                goalMotions_.insert(goalMotions_.end(), newBatchGoalMotions_.begin(), newBatchGoalMotions_.end());
+                newBatchGoalMotions_.clear();
+            }
+            OMPL_INFORM("%s: Increased time bound factor to %.2f", getName().c_str(), newBatchTimeBoundFactor);
+            continue;
+        }
+
+        // determine whether the old or new batch is sampled
+        sampleOldBatch_ =
+            (firstBatch || isTimeBounded_ || !sampleUniformForUnboundedTime_ || rng_.uniform01() <= oldBatchSampleProb);
+
+        ompl::base::State *goalState{nullptr};
+        if (sampleOldBatch_)
+        {
+            // sample until successful or time is up
+            if (goalMotions_.empty() && isTimeBounded_)
+                goalState = nextGoal(ptc, oldBatchTimeBoundFactor, newBatchTimeBoundFactor);
+            // sample for n tries, with n = batch size
+            else if (goalMotions_.empty() && !isTimeBounded_)
+            {
+                goalState = nextGoal(static_cast<int>(batchSize), oldBatchTimeBoundFactor, newBatchTimeBoundFactor);
+                // the goal region is most likely blocked for this time period -> increase upper time bound
+                if (goalState == nullptr)
+                {
+                    newBatchTimeBoundFactor *= timeBoundFactorIncrease_;
+                    oldBatchTimeBoundFactor = newBatchTimeBoundFactor;
+                    startTree = true;
+                    if (sampleUniformForUnboundedTime_)
+                        batchSize = std::ceil(2.0 * (timeBoundFactorIncrease_ - 1.0) *
+                                              static_cast<double>(tStart_->size() + tGoal_->size()));
+                    numBatchSamples = 0;
+                    OMPL_INFORM("%s: Increased time bound factor to %.2f", getName().c_str(), newBatchTimeBoundFactor);
```

Comment:
If possible, I would break coherent chunks of code into separate functions. Here you could use "IncreaseUpperTimeBound". This makes the code easier to read.